api/envoy/admin/v3/config_dump.proto

```diff
@@ -61,6 +62,14 @@ message UpdateFailureState {
   // This is the version of the rejected resource.
   // [#not-implemented-hide:]
   string version_info = 4;
+
+  // This resource is not included in the update or being explicitly removed by
+  // an update, either case means that this resource no longer existed in the
+  // xDS server. For more information, please refer to the :ref:`"Knowing When a
+  // Requested Resource Does Not Exist" <xds_protocol_resource_not_existed>`
+  // section.
+  // [#not-implemented-hide:]
+  bool does_not_exist_at_server = 5;
```

Comment:
Instead of adding a bool, how about making this an enum?  That way, we can differentiate between (e.g.) "recently subscribed but not yet received" and "resource does not exist on server".  We can also explicitly indicate some of the other existing states from the top-level enum, which is basically useless at that level.

Reply:
Adding an extra status enum feels duplicated with [`ClientConfigStatus`](https://github.com/envoyproxy/envoy/blob/main/api/envoy/service/status/v3/csds.proto#L59) in CSDS. What you think about the alternative to promote the `ClientConfigStatus` to `config_dump.proto`?

CSDS can either drop that field (since it can be deducted from per-resource state) or use the `ClientConfigStatus` in `config_dump.proto`.

Reply:
I'm fine with removing the top-level `ClientConfigStatus` and moving it here.  It's really not useful as a top-level field.

Reply:
The `ClientConfigStatus` is moved into `config_dump.proto` and it is renamed to `ConfigErrorStatus` to fit it's parent message name `UpdateFailureState`.